WordPress/Classes/ViewRelated/System/Bottom Sheet/BottomSheetContentViewController.swift

```diff
@@ -0,0 +1,113 @@
+import UIKit
+import SwiftUI
+import WordPressUI
+
+class BottomSheetContentViewController: UIViewController {
+
+    // MARK: - Views
+
+    private let scrollView: UIScrollView = {
+        let view = UIScrollView()
+        view.showsVerticalScrollIndicator = false
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private let contentView: UIView
+
+    private var hostingController: UIViewController?
+
+    // MARK: - Init
+
+    init(contentView: UIView) {
+        self.contentView = contentView
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    init<T: View>(contentView: T) {
+        let hostingController = UIHostingController(rootView: contentView)
+        self.contentView = hostingController.view
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        if let hostingController {
+            self.setupHostingController(hostingController)
+        } else {
+            self.setupContentView()
+        }
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        self.updateContentSize()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        guard previousTraitCollection?.preferredContentSizeCategory != traitCollection.preferredContentSizeCategory else {
+            return
+        }
+        self.view.setNeedsLayout()
+        self.view.layoutIfNeeded()
+        self.updateContentSize()
+        if let presentationController = parent?.presentationController as? DrawerPresentationController {
+            presentationController.transition(to: presentationController.currentPosition)
+        }
+    }
```

Comment:
This code adjusts the layout of the bottom sheet when fonts are enlarged. I needed a way to update the parent view in response to changes in the child view's size. While I don't like this approach of accessing `DrawerPresentationController` due to it making too many assumptions about implementation details, it is the best approach I could think of.

Reply:
Do we need to handle the `else` path here? I think it is fine as long as it works with other VC types. Otherwise, it would make sense to specify the type rather than make it look like it is more flexible.